git/operations.go

```diff
@@ -41,11 +42,31 @@ func clone(ctx context.Context, workingDir string, keyRing ssh.KeyRing, repoURL,
 	return repoPath, nil
 }
 
-func commit(ctx context.Context, workingDir, commitMessage string) error {
+func commit(ctx context.Context, workingDir string, commitAction *CommitAction) error {
+	commitAuthor := commitAction.Author
+	// The user in the update.Cause.User must be a valid github user, otherwise the commit fails
```

Comment:
This is not quite accurate. git (not github) will reject a commit if you supply only an email address rather than e.g., "Michael Bridgen <michael@weave.works>", and that email address has not been used in a prior revision.

This is certainly a wrinkle though. Some alternatives:

assume that commits will have been made by a particular user, and let git reject those that aren't
make people supply a git-author-like value as the username, e.g., fluxctl release --user="Michael Bridgen <michael@weave.works>" .... Not so easy if you are the web UI and you simply don't have all of that information.
let people toggle this behaviour, and then do one of the above in the case that it's switched on

Reply:
These three versions worked:

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="Michael Bridgen <michael@weave.works>" lock --service=default/helloworld
SERVICE             STATUS   UPDATES
default/helloworld  success  
Commit pushed:	af4ed19

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="Michael Bridgen" unlock --service=default/helloworld
SERVICE             STATUS   UPDATES
default/helloworld  success  
Commit pushed:	40f30d4

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="michael@weave.works" lock --service=default/helloworld
SERVICE             STATUS   UPDATES
default/helloworld  success  
Commit pushed:	036e2a5

Reply:
Yup; the first gives the full author format, so it succeeds (for the record: there are no commits using that name _and_ email as the author in `weaveworks/flux-example`). The second and third succeed because git can find a commit with the username, and email, respectively. If you look at the commit log, it has the full author user and email recorded.

Reply:
This does not work:

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="test@test" unlock --service=default/helloworld
Error: git commit: fatal: --author 'test@test' is not 'Name <email>' and matches no existing author

and this does:

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="Tamara Kaufler <test@test>" unlock --service=default/helloworld
SERVICE             STATUS   UPDATES
default/helloworld  success

Reply:
The potential "unhappy path" is if you supply an email or a user (but not both) that hasn't been used before. For example,

    fluxctl --url=http://$(minikube ip):30081/api/flux --token=local-test-token --user="Brichael Midgen" unlock --service=default/helloworld

Reply:
These work as well:

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="Test <test@test>" lock --service=default/helloworld

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="Test <test>" unlock --service=default/helloworld

fluxctl --url=http://$(minikube ip):30081/api/flux  --token=local-test-token --user="test@test <test@test>" lock --service=default/helloworld


The custom author will be accepted if the format is:
some_text <another_text> 
some_text<some_text>